.github/workflows/uv-lock.yml

```diff
@@ -0,0 +1,24 @@
+# pre-commit.ci does not allow network access, so we run this on github actions
+name: Update uv.lock
+
+on:
+  pull_request:
+  pull_request_target:
+  push:
+    branches: [develop]
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        ref: ${{ github.head_ref }}
+    - name: Install uv
+      uses: astral-sh/setup-uv@v5
+    - name: Update lock
+      run: uv lock
```

Comment:
`uv lock` should be run on the merge commit between the PR and the base branch (that's how pre-commit.ci does it), rather than running `uv lock` on the PR's head commit. Running it on the head commit might result in a situation where all tests pass for the PR, but fail after the merge.